clippy_lints/src/matches.rs

```diff
@@ -121,6 +121,26 @@ declare_lint! {
     "a match with overlapping arms"
 }
 
+/// **What it does:** Checks for arm matches all errors with `Err(_)`.
+///
+/// **Why is this bad?** It is a bad practice to catch all errors the same way
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// let x : Result(i32, &str) = Ok(3);
+/// match x {
+///     Ok(_) => println!("ok"),
+///     Err(_) => println!("err"),
```

Comment:
yes, we might extend it to more things later, but for now, just if there's a `panic!` (could probably generalize it to anything of type `!`, not sure how far that is propagated yet.)